src/main.rs

```diff
@@ -31,6 +42,27 @@ struct UsageError {
     message: String,
 }
 
+/// Parses the value of a flag that takes a duration.
+///
+/// At the moment, the only supported unit for the duration are seconds and the textual value
+/// must include the "ms" suffix to support adding extra units in the future if desired.
+fn parse_duration(value: &str) -> Result<Timespec, UsageError> {
+    if !value.ends_with(SECONDS_SUFFIX) {
+        let message = format!("invalid time specification {}: must end in 's'", value);
+        return Err(UsageError { message });
+    }
+    let value = &value[0..value.len() - SECONDS_SUFFIX.len()];
```

Comment:
I would add parens because I feel that having no space after the `..` makes it bind more tightly in my brain so I prefer to include them. You can also omit the lower bound.
```suggestion
    let value = &value[..(value.len() - SECONDS_SUFFIX.len())];
```

Reply:
Not applicable any longer.